asset_dashboard/static/js/components/maps/AssetsMap.js

```diff
@@ -0,0 +1,177 @@
+import ReactDOM from 'react-dom'
+import React, { useState, useEffect } from 'react'
+import { GeoJSON } from 'react-leaflet'
+import hash from 'object-hash'
+import Cookies from 'js-cookie'
+import BaseMap from './BaseMap'
+import AreaClipper from '../map_utils/AreaClipper'
+import AssetSearchTable from '../tables/AssetSearchTable'
+
+function AssetTypeOptions() {
+  // these options could come from the server but hardcoding for now 
+  const options = [
+    {value: 'buildings', label: 'Buildings'},
+    {value: 'trails', label: 'Trails'}
+  ]
+
+  return options.map(option => {
+    return <option value={option.value} key={option.value}>{option.label}</option>
+  })
+}
+
+function AssetsMap(props) {
+  const [searchedGeoms, setSearchedGeoms] = useState()
+  const [existingGeoms, setExistingGeoms] = useState()
+  const [clippedGeoms, setClippedGeoms] = useState(null)
+  const [searchText, setSearchText] = useState('')
+  const [searchedAssetType, setSearchedAssetType] = useState('buildings')
+
+  useEffect(() => {
+    if (props?.existing_geoms) {
+      setExistingGeoms(JSON.parse(props.existing_geoms))
+    }
+  }, [setSearchedGeoms, setExistingGeoms])
+
+  function onMapCreated(map) {
+    const group = new L.featureGroup()
+
+    map.eachLayer((layer) => {
+      if (layer.feature) {
+        group.addLayer(layer)
+      }
+    })
+
+    if (Object.keys(group._layers).length > 0) {
+      map.fitBounds(group.getBounds())
+    }
+  }
+
+  function onClipped(featureCollection) {
+    setClippedGeoms(featureCollection)
+  }
+
+  function saveGeometries() {
+    fetch(window.location.pathname, {
+        headers: {
+            'Accept': 'application/json',
+            'Content-Type': 'application/json',
+            'X-CSRFTOKEN': Cookies.get('csrftoken')
+        },
+        method: 'POST',
+        mode: 'same-origin',
+        body: JSON.stringify({
+          'geojson': clippedGeoms,
+          'phase': selectedPhase.value
+        })
+    }).then((response) => {
+      if (response.status == 200) {
+        location.reload()
+        // TODO: show success message
+        // https://stackoverflow.com/questions/13256817/django-how-to-show-messages-under-ajax-function
+      } else {
+        // TODO: catch error and show message
+      }
+    })
+  }
+
+  function searchAssets() {
+    const url = `/assets/?` + new URLSearchParams({
+      'q': searchText,
+      'asset_type': searchedAssetType
+    })
+
+    fetch(url, {
+      headers: {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json',
+        'X-CSRFTOKEN': Cookies.get('csrftoken')
+      },
+      mode: 'same-origin',
+      method: 'GET'
+    }).then((response) => response.json())
+    .then((data) => setSearchedGeoms(data))
```

Comment:
Can we zoom in on search results easily here?

Reply:
Not easily. The first iteration of this worked like that because I was loading the search results from the django template. When the component first loads, the [whenCreated callback executes in the BaseMap component](https://github.com/fpdcc/ccfp-asset-dashboard/blob/4d384825e96be01b749140161b2f8d31ef6be1d7/asset_dashboard/static/js/components/maps/AssetsMap.js#L155) and any geojson that's loaded from props will exist as a layer in the map `whenCreated`. Therefore the map will [zoom to fit those geojson layer](https://github.com/fpdcc/ccfp-asset-dashboard/blob/4d384825e96be01b749140161b2f8d31ef6be1d7/asset_dashboard/static/js/components/maps/AssetsMap.js#L45). 

Fetching the search results from a json api broke the zooming upon search, but it will still work when the page loads and there are existing assets. I tried to fix this but it was a bit more involved, so I decided to get this PR in shape and address that later.

Reply:
Good call, thanks.

Reply:
might be able to create a bounding box with django-rest-framework-gis? saving here for later reference: https://github.com/openwisp/django-rest-framework-gis#bounding-box-auto_bbox-and-bbox_geo_field